app/controllers/teams_controller.rb

```diff
@@ -59,55 +55,82 @@ def destroy
 
   # Assigns team ownership to a specific team member
   def assign_ownership
-    unless owners_include_user
-      team_owners << user
-      redirect_to @team
-    end
+    assign_ownership_to_member
+    redirect_to @team
   end
 
   def delete_ownership
-    delete_owner_if_existing
+    delete_ownership_from_member
     redirect_to @team
   end
 
   def delete_membership
-    delete_owner_if_existing
-    @team.members.delete(user)
+    delete_membership_from_member
     redirect_to @team
   end
 
-  def current_ability
-    @current_ability ||= Ability.new(current_user, params[:team_member])
+  def perform_action_on_multiple_members
+    target_members = params[:members]
+    current_user_id_string = current_user.id.to_s
+    if target_members.include? current_user_id_string
+      target_members = (target_members - [current_user_id_string]) + [current_user_id_string]
```

Comment:
This line logically does not make sense to me. Why are you deleting the current user from the members before you add him in the same way? Could you please explain?

Reply:
done so that `current_that`is last in array and thus last to be performed action on

Reply:
Alright, thanks! :smile_cat: